src/join_reference.rs

```diff
@@ -0,0 +1,57 @@
+use std::fmt::{Debug, Display, Formatter};
+use std::sync::Arc;
+
+use crate::reference::Reference;
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
+pub struct JoinReference(Reference);
+impl JoinReference {
+    pub fn new() -> Self {
+        Self(Reference::new())
+    }
+}
+impl Debug for JoinReference {
```

Comment:
This could be derived

Reply:
The derived version of `Debug` with `alternate` true, that is the `{:#?}` format specificier produces a pretty-printed format that still contains the name of the struct instead of just being the internal string:

```rust
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn join_reference_debug_without_alternate() {
        let join_reference: JoinReference = "join_reference".into();

        assert_eq!(format!("{:?}", join_reference), "JoinReference(Reference(\"join_reference\"))")
    }

    #[test]
    fn join_reference_debug_with_alternate() {
        let join_reference: JoinReference = "join_reference".into();

        assert_eq!(format!("{:#?}", join_reference), "\"join_reference\"")
    }
}
```

```
thread 'join_reference::tests::join_reference_debug_with_alternate' panicked at 'assertion failed: `(left == right)`
  left: `"JoinReference(\n    \"join_reference\",\n)"`,
 right: `"\"join_reference\""`', src/join_reference.rs:72:9
```

So, the `Debug` as written is not equivalent to the `derive`d one.